source/Irrlicht/COpenGLDriver.cpp

```diff
@@ -1862,7 +1862,9 @@ void COpenGLDriver::draw2DLine(const core::position2d<s32>& start,
 
 		glDrawElements(GL_LINES, 2, GL_UNSIGNED_SHORT, Quad2DIndices);
 
-		// Draw non-drawn last pixel (search for "diamond exit rule")
+		// Draw sometimes non-drawn first & last pixel (search for "diamond exit rule")
+		// HACK this messes with alpha blending
+		glDrawArrays(GL_POINTS, 0, 0);

```

Comment:
If I change the third parameter to `1`, the issue is fixed.

```suggestion
		glDrawArrays(GL_POINTS, 0, 1);

```

Reply:
Should've read the gl docs instead of making the stupid assumption that these were start & end indices (both inclusive):

> void glDrawArrays( | GLenum | mode, | GLint | first, | GLsizei | count);

https://docs.gl/gl3/glDrawArrays